### DISCUSSION TITLE
Force Rediscovery in HomeAssistant?

### DISCUSSION THREAD

**Discussion by stephenmetzger** [mKeRix/room-assistant#428] [Q&A]
Hello, 

I recently got a new phone and tried to update my room-assistant configuration to use it's bluetooth classic MAC address to replace my old phone's config. I thought it would be as easy as swapping out the MAC address in the local.yaml files on the three instances of R-A I have running, but its turning out to be more challenging. 

I see the mqtt messages coming through for the topic of the new phone, but it the existing homeassistant entity was not reflecting the changes I was seeing in the mqtt messages. So, I deleted the entity from home-assistant assuming that it would be re-discovered when I restarted R-A, but no dice.  auto-discovered in homeassistant. Is there any way to force rediscovery?

Thanks!

**Comment by mKeRix** [comment]
Re-discovering an auto-discovered MQTT entity in Home Assistant can usually be done by just restarting Home Assistant. Switching to a new phone should really be that easy, swap out the MACs and then use the newly created entities in Home Assistant instead (or change the entity IDs around so there so that they match your old ones).

After each start of room-assistant the instance will send auto discovery information to the topic the first time it sees a Bluetooth device.

**Reply by stephenmetzger** [reply]
Thanks @mKeRix. Unfortunately, this doesn't seem to be the case for me. I've implemented the previously deleted entity for the problematic bluetooth device (my iPhone)  as a device_tracker in homeassistant, which is working as expected, but it is still not being discovered in hassio for room-tracking purposes.

**Reply by mKeRix** [reply]
Can you double-check that the auto-discovery is turned on for the MQTT integration in Home Assistant? And that there are no error messages in the logs regarding it? I mean it worked before and should work again now, but this way we can ensure that this isn't just some crazy one-off bug that changed your settings in HA by accident.
